shared/interfaces/metadata_wrapper.go

```diff
@@ -0,0 +1,101 @@
+package interfaces
+
+import (
+	"github.com/prysmaticlabs/go-bitfield"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// MetadataV1 is a convenience wrapper around our metadata protobuf object.
```

Comment:
```suggestion
// MetadataV0 is a convenience wrapper around our metadata protobuf object.
```

I wonder whether using `V0` and `V1` is better given we mostly start with versioning 0. `pb.MetaData` without any versioning makes me think it's closer to 0 than to 1 🤔

Reply:
good question lol, would technically have to change every reference in that case. I can try and changing it to a 0 and 1 to see how much work it would take.